contracts/rate_oracles/LinearRateOracle.sol

```diff
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier: Apache-2.0
+
+pragma solidity =0.8.9;
+
+import "./OracleBuffer.sol";
+import "./BaseRateOracle.sol";
+import "../interfaces/rate_oracles/IRateOracle.sol";
+import "../core_libraries/FixedAndVariableMath.sol";
+import "prb-math/contracts/PRBMathUD60x18.sol";
+import "../interfaces/IFactory.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "../core_libraries/Time.sol";
+import "../utils/WadRayMath.sol";
+
+/// @notice Contract base for a Rate Oracle with linear rates implementation.
+///  This contract is abstract. To make the contract deployable override the
+/// `getLastUpdatedRate` function and the `UNDERLYING_YIELD_BEARING_PROTOCOL_ID` constant.
+/// @dev Each specific rate oracle implementation will need to implement the virtual functions
+abstract contract LinearRateOracle is BaseRateOracle {
+    using OracleBuffer for OracleBuffer.Observation[65535];
+
+    /// @inheritdoc BaseRateOracle
+    function getGrowthBetween(uint256 rateFromRay, uint256 rateToRay)
+        internal
+        pure
+        override
+        returns (uint256 growthBetweenRay)
+    {
+        return rateToRay - rateFromRay;
+    }
+
+    /// @notice Computes the APY based on the un-annualised rateFromTo value and timeInYears (in wei)
+    /// @param rateFromToWad Un-annualised rate (in wei)
+    /// @param timeInYearsWad Time in years for the period for which we want to calculate the apy (in wei)
+    /// @return apyWad APY for a given rateFromTo and timeInYears
+    function computeApyFromRate(uint256 rateFromToWad, uint256 timeInYearsWad)
+        internal
+        pure
+        override
+        returns (uint256 apyWad)
+    {
+        if (rateFromToWad == 0) {
+            return 0;
+        }
+
+        uint256 exponentWad = PRBMathUD60x18.div(
```

Comment:
no longer an exponent right given we're just multiplying?

Reply:
true, called it multiplier [c2a5c2f](https://github.com/Voltz-Protocol/voltz-core/pull/230/commits/c2a5c2fda8ffd7f8ede64ea6b3b1d51a822d32d0)